rmc/utils/mpc.py

```diff
@@ -433,51 +433,78 @@ def _run_glm(
         min_single_aic_var,
     )
 
-    # List possible variable combinations for Joint Regression
-    comb_var = []
-    for k in range(2,6):
-        comb_var += [x for x in combinations(all_var,k)]
+    # List possible variable combinations for joint regression
+    var_combs = []
+    for k in range(2, len(all_var) + 1):
+        var_combs += [x for x in combinations(all_var, k)]
 
-    logger.info("Running joint (additive interactions only) regression...")
     add_model_res = {}
     add_model_aic = []
-    add_X = []
-    for var_a in comb_var:
-        logger.info("Running joint (additive interactions) regression for %s...", var_a)
-        formula = f"pop_v_path ~ {' + '.join(var_a)}"
+    add_model_X = []
+    for var_comb_add in var_combs:
```

Comment:
I would prefer just calling this `var_comb` (same with `var_comb_mult` -> `var_comb`)

```suggestion
    for var_comb in var_combs:
```

Reply:
I thought about it but it was failing mypy (it was surviving the previous for loop) hence I changed it.

Reply:
Actually never mind about this

Reply:
Yes that is the plan. I'm working out a helper function which would make the code simpler as well as avoid mypy issues. I really liked your suggestion about extracting a function :)